tools/code_analyzer/op_deps_pass.cpp

```diff
@@ -744,8 +744,12 @@ class OpDependency : public ModulePass {
       const std::string& contextualNamespace, Value* V) {
     std::vector<std::string> schemaStrs;
     extractStringValue(V, [&](const std::string& str) {
+      // NB: when using TORCH_SELECTIVE_NAME, we will conventionally provide
+      // the ENTIRE string (namespace included).  If this occurs, we MUST NOT
+      // use the contextual namespace.  Fortunately, it's easy to tell if
+      // a namespace is included: a double colon will be present.
       const std::string& schemaStr =
-          contextualNamespace.empty() ? str : contextualNamespace + str;
+          (contextualNamespace.empty() || str.find("::") != std::string::npos) ? str : contextualNamespace + str;
```

Comment:
lgtm - just to confirm, so contextual namespace and operator namespace will both exist after moving to the new world?

Reply:
Yes, although it would be nice if the constexpr machinery could get the contextual namespace so I could get rid of this https://github.com/pytorch/pytorch/pull/39401#issuecomment-639599944